src/backend/distributed/operations/repair_shards.c

```diff
@@ -849,6 +849,14 @@ CopyShardTables(List *shardIntervalList, char *sourceNodeName, int32 sourceNodeP
 
 		List *ddlCommandList = CopyShardCommandList(shardInterval, sourceNodeName,
 													sourceNodePort, includeDataCopy);
+
+		if (PartitionTable(shardInterval->relationId))
+		{
+			char *attachPartitionCommand =
+				GenerateAttachShardPartitionCommand(shardInterval);
+
+			ddlCommandList = lappend(ddlCommandList, attachPartitionCommand);
+		}													
```

Comment:
should this be in a separate loop such that it happens after creating all the shards?